script.js

```diff
@@ -0,0 +1,236 @@
+// Load data from a JSON file
+d3.json("hierarchy.json").then((data) => {
+  // Specify the chart’s dimensions
+  const width = 928;
+  const height = width;
+  const radius = width / 6;
+
+  // Create the color scale
+  const color = d3.scaleOrdinal(
+    d3.quantize(d3.interpolateRainbow, data.children.length + 1)
+  );
+
+  // Compute the layout
+  const hierarchy = d3
+    .hierarchy(data)
+    .sum((d) => d.value)
+    .sort((a, b) => b.value - a.value);
+  const root = d3.partition().size([2 * Math.PI, hierarchy.height + 1])(
+    hierarchy
+  );
+  root.each((d) => (d.current = d));
+
+  // Create the arc generator
+  const arc = d3
+    .arc()
+    .startAngle((d) => d.x0)
+    .endAngle((d) => d.x1)
+    .padAngle((d) => Math.min((d.x1 - d.x0) / 2, 0.005))
+    .padRadius(radius * 1.5)
+    .innerRadius((d) => d.y0 * radius)
+    .outerRadius((d) => Math.max(d.y0 * radius, d.y1 * radius - 1));
+
+  // Create the SVG container
+  const svg = d3
+    .select("#chart")
+    .append("svg")
+    .attr("viewBox", [-width / 2, -height / 2, width, height])
+    .style("font", "10px sans-serif");
+
+  // Append the arcs
+  const path = svg
+    .append("g")
+    .selectAll("path")
+    .data(root.descendants().slice(1))
+    .join("path")
+    .attr("fill", (d) => {
+      while (d.depth > 1) d = d.parent;
+      return color(d.data.name);
+    })
+    .attr("fill-opacity", (d) =>
+      arcVisible(d.current) ? (d.children ? 0.6 : 0.4) : 0
+    )
+    .attr("d", (d) => arc(d.current));
+
+  // ---------- chakra playback -------------
+
+  let currentChakraAudio = null; // Variable to keep track of currently playing chakra audio
+  let currentDiseaseAudio = null; // Variable to keep track of currently playing disease audio
+
+  // Function to extract chakra name from the string
+  function extractChakraName(name) {
+    const match = name.match(/\(([^)]+)\)/); // Regular expression to find text in parentheses
+    return match ? match[1] : null; // Return the matched group or null if no match
+  }
+
+  // Function to play chakra sound
+  function playChakraSound(chakraName) {
+    console.log("Playing chakra sound for:", chakraName);
+    const filePath = `/public/chakra_sounds_mp3/${chakraName}.mp3`;
+    if (currentChakraAudio) {
+      currentChakraAudio.pause();
+      currentChakraAudio.currentTime = 0;
+    }
+    currentChakraAudio = new Audio(filePath);
+    currentChakraAudio.loop = true;
+    currentChakraAudio.play().catch((e) => {
+      console.error("Failed to play chakra audio:", e);
+    });
+  }
+
+  // ---- Disease sound ----------
+  let diseaseData = []; // This will store the parsed CSV data
+
+  // Function to load and parse the CSV
+  function loadDiseaseData() {
+    d3.csv("/public/seq.d3.csv").then((data) => {
+      diseaseData = data;
+    });
+  }
+
+  // Call the function to load the data
+  loadDiseaseData();
```

Comment:
Unless I misunderstood, I'd rather not rely on the CSV just for row numbers to identify which audio file to play. We can encode this mapping in the JSON instead.